design/ceph-config-updates.md

```diff
@@ -0,0 +1,217 @@
+- `--keyring` may necessary to inform daemons where to find the keyring which must be mounted to a
+  nonstandard directory by virtue of being sourced via a Kubernetes secret.
+
+Notable non-required flags:
+- `--{mon,osd,mgr,mds,rgw}-data-dir` settings exist for all daemons, but it is more desirable to use
+  the `/var/lib/ceph/<daemon_type>/ceph-<daemon_id>` directory for daemon data within containers. If
+  possible, mapping the `dataDirHostPath/<rook_daemon_data_dir>` path on hosts to this default
+  location in the container is preferred.
+- `--run-dir` exists for all daemons, but it is likewise more desirable to use the `/var/run/ceph`
+  path in containers. If persisting run dir data to the host is required, mapping the
+  `dataDirHostPath/<rook_daemon_run_dir>` path on hosts to this default location is preferred;
+  however, this may be practically tricky for the mons since the on-disk location of the mon run dir
+  is `dataDirHostPath/mon-<id>` and the on-disk location of mon data is
+  `dataDirHostPath/mon-<id>/data`, which is the same as saying `<mon_run_dir>/data`.
+  **TODO** does run dir data need persisted to the host? If not, this can be stated instead, and we
+  will propose to delete any preexisting run data stored in the data dir and simply use the
+  `/var/run/ceph` location in containers for runtime storage of the data.
+
+### Additional configuration
+
+Additional configuration which Rook sets up initially should be done by setting values in Ceph's
+centrally-stored config. A large group of additional configurations can be configured at once via
+the Ceph command `ceph config assimilage-conf`. Care should be taken to make sure that Rook does not
+modify preexisting user-specified values.
+**TODO** how to determine if a user has specified a value or if it is at default state?
+
+Existing global configs configured here:
+- `mon allow pool delete = true`
+- `fatal signal handlers = false`
+  **TODO** here unless it is required by virtue of the container context as with `log file`
+- `log stderr prefix = "debug "` should be set for all daemons to differentiate logging from auditing
+- `debug ...` configs
+
+Removed configs:
+- `mon_max_pg_per_osd = 1000` is a dangerous setting and should be removed regardless of whether
+  this proposal is accepted
+- `mon keyvaluedb = rocksdb` is not needed for Luminous+ clusters
+- `filestore_omap_backend = rocksdb` is not needed for Luminous+
+- `osd pg bits = 11` set using config override for testing or play clusters ???
+- `osd pgp bits = 11` set using config override for testing or play clusters ???
+- `osd pool default size = 1` set using config override for testing or play clusters
+- `osd pool default min size = 1` set using config override for testing or play clusters
+- `osd pool default pg num = 100` set using config override for testing or play clusters ???
+- `osd pool default pgp num = 100` set using config override for testing or play clusters ???
+- `rbd_default_features = 3`
```

Comment:
@dillaman we can drop this, right?

Reply:
Assuming this is for k8s RBD persistent volume dynamic provisioner, this would be needed since krbd doesn't support the full range of default RBD image features.

Reply:
If I understand the clarification, if we use k8s' RBD PV dynamic provisioner, we do need this. But if the Ceph CSI driver has its own provisioner, it's possible we no longer need this. My gut reaction is to keep this, because I doubt we will be able to stop users from using the k8s RBD PV dynamic provisioner. I'd love a definitive answer here though; can we 'force' use of the Ceph CSI driver for this? Does the Ceph CSI driver even implement this since it already exists in k8s?

Reply:
I believe the RBD CSI will, by default, pass an empty list of features to the "rbd create" call -- which will result in a krbd-compatible image by default. In that case, this wouldn't be needed within the k8s  controller.

Reply:
As long as you are using k8s 1.8 or later, it should actually properly handle the image features w/o the need to override.